### PR TITLE
MAINT: Removing deprecated `offset_position` from Collection

### PR DESCRIPTION
**Pull request by greglucas**
## PR Summary

This removes the 3.3 deprecation of `offset_position` from the Collection class.

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by anntzer** [comment]
There's also a fair amount of corresponding support that should be removed on the C++ side.

**Comment by jklymak** [comment]
@greglucas  if the C++ code is more than you planned to bite off, we could probably merge this and have a new PR for the C++ code.  Do we actually need to remove the C++ code?

**Comment by anntzer** [comment]
We "don't", just like we don't need to remove any other unused code, but we probably should.  That can be done as a separate PR, though; I just mentioned it here so that it doesn't get forgotten.

**Review by jklymak — APPROVED** [review]

**Comment by greglucas** [comment]
How big of a hammer can we use for it? Seems like most of the code removal would require changing the number of arguments to the public backend function calls.

i.e. `draw_path_collection()` would have one less arg required.

I'm all for cleaning up the code, but that may cause headaches for other backend implementers?

**Comment by greglucas** [comment]
The latest commit covers the full removal from the backends. Happy to drop that if it is a little too much.

**Comment by QuLogic** [comment]
I'm not sure that's quite what the deprecation said; only `offset_position='screen'` is deprecated, not the parameter itself.

**Comment by greglucas** [comment]
Well, there were previously only two possibilities in that enum. So you're really just passing a constant around now I think?

**Comment by QuLogic** [comment]
Dropping the argument in the caller would break third-party backend implementations that did not have it optional.

**Comment by greglucas** [comment]
OK, I figured that may be problematic. I'll try again here by leaving the unused argument in place and removing all the other downstream AGG code that used it.

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -90,7 +89,6 @@ def __init__(self,
                  pickradius=5.0,
                  hatch=None,
                  urls=None,
-                 offset_position='screen',
```
Comment (QuLogic) [inline lib/matplotlib/collections.py:93]: `zorder` becomes keyword-only after this removal.

File: doc/api/next_api_changes/removals/20245-GL.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Collection's offset_position has been removed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The deprecated keyword argument ``offset_position`` has been removed from
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/20245-GL.rst:4]:
```suggestion
The deprecated keyword argument *offset_position* has been removed from
```

**Review by QuLogic — APPROVED** [review]

**Pull request merged**